File: python/src/hywrapper/models/other/punishment_stats_response.py
from __future__ import annotations

from typing import Optional

from pydantic import Field

from hywrapper.models.hypixel_response import HypixelResponse


class PunishmentStatsResponse(HypixelResponse):
    watchdog_last_minute: Optional[int] = Field(default=None, alias="watchdog_lastMinute")
    staff_rolling_daily: Optional[int] = Field(default=None, alias="staff_rollingDaily")
    watchdog_total: Optional[int] = Field(default=None, alias="watchdog_total")
    watchdog_rolling_daily: Optional[int] = Field(default=None, alias="watchdog_rollingDaily")
    staff_total: Optional[int] = Field(default=None, alias="staff_total")

File: typescript/src/models/other/PunishmentStatsResponse.ts
import { HypixelResponse } from "../../types";

export interface PunishmentStatsResponse extends HypixelResponse {
    watchdog_lastMinute?: number;
    staff_rollingDaily?: number;
    watchdog_total?: number;
    watchdog_rollingDaily?: number;
    staff_total?: number;
}
